### DISCUSSION TITLE
Connecting to a Sites.Selected site fails with the message "The app principal does not exist"

### DISCUSSION THREAD

**Discussion by plamber** [pnp/powershell#3266] [Q&A]
Hi, 
we assigned to an Azure AD APP with the ID and Sites.Selected application permissions the permissions using PnP PowerShell.

```
$siteUrl = 'https://yourTenantName.sharepoint.com/sites/o-000401'
$targetApplicationID = "yourAppId"

Connect-PnPOnline $siteUrl -Interactive
Grant-PnPAzureADAppSitePermission -AppId $targetApplicationID -DisplayName 'appsitepermissions' -Site $siteUrl -Permissions Write

Disconnect-PnPOnline
```

When trying to connect using the app I am receiving this error message.
```
get-pnpsite : Cannot contact site at the specified URL https://yourTenantName.sharepoint.com/sites/o-000401. The app principal does not exist.
```
when running the script
```
$ClientId = "yourAppId"
$ClientSecret = "yourSecret"
$siteUrl = 'https://yourTenantName.sharepoint.com/sites/o-000401'

Connect-PnPOnline $siteUrl -ClientId $ClientId -ClientSecret $ClientSecret
get-pnpsite
```

Have you seen this error and do you happen to know how to fix this?

Thank you

**Comment by gautamdsheth** [comment]
Hi @plamber , would recommend using certificate parameters for authentication. The client-id/secret uses the old ACS flow. Also , if you do indeed need to use client secret then I guess tenant parameter is necessary. Check the docs for connect-pnponline

**Comment by plamber** [comment]
HI @gautamdsheth,
thank you for reaching out. 

Which tenant parameter are you referring to? I am also going to give it a try with the certificate.

br,
Patrick
